fedimint-cli/src/client.rs

```diff
@@ -421,10 +421,23 @@ pub async fn handle_command(
             }))
         }
         ClientCmd::Withdraw { amount, address } => {
-            let fees = client.get_withdraw_fee(address.clone(), amount).await?;
-            let absolute_fees = fees.amount();
+            let (amount, fees) = match amount {
+                // If the amount is "all", then we need to subtract the fees from
+                // the amount we are withdrawing
+                BitcoinAmountOrAll::All => {
+                    let balance =
+                        bitcoin::Amount::from_sat(client.get_balance().await.msats * 1000);
+                    let fees = client.get_withdraw_fee(address.clone(), balance).await?;
+                    (balance - fees.amount(), fees)
+                }
+                BitcoinAmountOrAll::Amount(amount) => (
+                    amount,
+                    client.get_withdraw_fee(address.clone(), amount).await?,
+                ),
+            };
+            let absolute_fee = fees.amount();
```

Comment:
Feels like this should be implemented on the wallet module so we don't duplicate the code in the gateway. #3573

Reply:
Moved into fedimint-wallet-common. Then if we want to refactor to have the generalized `AllOr<T>` we can do that in fedimint-core but for now wallet sounds good.

Reply:
gateway doesn't have fedimint-wallet as a dependency, cool to add it and move to fedimint-wallet or should I keep it in fedimint core and make it generic `AllOr<T>` like @Maan2003 suggests?

Reply:
I see, then let's leave it here